Models/ImageClassification/DenseNet.swift

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TensorFlow
+
+// Original Paper:
+// Densely Connected Convolutional Networks
+// Gao Huang, Zhuang Liu, Laurens van der Maaten, Kilian Q. Weinberger
+// https://arxiv.org/pdf/1608.06993.pdf
+
+public struct DenseNet: Layer {
+    public var conv = Conv(
+        filterSize: 7,
+        stride: 2,
+        inputFilterCount: 3,
+        outputFilterCount: 64
+    )
+    public var maxpool = MaxPool2D<Float>(
+        poolSize: (3, 3),
+        strides: (2, 2),
+        padding: .same
+    )
+    public var denseBlock1 = DenseBlock(repitionCount: 6, inputFilterCount: 64)
+    public var transitionLayer1 = TransitionLayer(inputFilterCount: 256)
+    public var denseBlock2 = DenseBlock(repitionCount: 12, inputFilterCount: 128)
+    public var transitionLayer2 = TransitionLayer(inputFilterCount: 512)
+    public var denseBlock3 = DenseBlock(repitionCount: 24, inputFilterCount: 256)
+    public var transitionLayer3 = TransitionLayer(inputFilterCount: 1024)
+    public var denseBlock4 = DenseBlock(repitionCount: 16, inputFilterCount: 512)
+    public var globalAvgPool = GlobalAvgPool2D<Float>()
+    public var dense: Dense<Float>
+
+    public init(classCount: Int) {
+        dense = Dense(inputSize: 1024, outputSize: classCount, activation: softmax)
+    }
+
+    @differentiable
+    public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+        let inputLayer = input.sequenced(through: conv, maxpool)
+        let level1 = inputLayer.sequenced(through: denseBlock1, transitionLayer1)
+        let level2 = level1.sequenced(through: denseBlock2, transitionLayer2)
+        let level3 = level2.sequenced(through: denseBlock3, transitionLayer3)
+        let output = level3.sequenced(through: denseBlock4, globalAvgPool, dense)
+        return output
+    }
+}
+
+extension DenseNet {
+    public struct Conv: Layer {
+        public var bNorm: BatchNorm<Float>
```

Comment:
Avoid abbreviations, especially single-character ones.
```suggestion
        public var batchNorm: BatchNorm<Float>
```